### PR TITLE
Events requested by Blockscout

### PR DESCRIPTION
**Pull request by DrZoltanFazekas**
The same events can also be used by the UCCB validator oracle to keep track of changes of the committee and the validators' stake. The events in this PR replace the `StakerAdded` and `StakerRemoved` events implemented in https://github.com/Zilliqa/zq2/pull/1178

### CODE REVIEW

File: zilliqa/src/contracts/deposit.sol [hunk 1]
```diff
@@ -421,6 +435,13 @@ contract Deposit {
         }
     }
 
+    // Returns the next block number at which new stakers are added,
+    // existing ones removed and/or deposits of existing stakers change
+    function nextUpdate() public view returns (uint256 blockNumber) {
+        if (latestComputedEpoch > currentEpoch())
+            blockNumber = latestComputedEpoch * blocksPerEpoch;
```
Comment (JamesHinshelwood) [inline zilliqa/src/contracts/deposit.sol:442]: Why is this not `blockNumber = (currentEpoch() + 2) * blocksPerEpoch`? That's the logic we use in `deposit` and `unstake` so shouldn't we be consistent?

**Review by JamesHinshelwood — APPROVED** [review]

File: zilliqa/src/contracts/deposit.sol (hunk 1 above)
Comment (DrZoltanFazekas) [reply zilliqa/src/contracts/deposit.sol:442]: Not really imo. If there are future changes then `latestComputedEpoch` determines when they become effective. This is either at the beginning of `currentEpoch() + 2` or if this function is called later, maybe already at the beginning of `currentEpoch() + 1`.

Comment (DrZoltanFazekas) [reply zilliqa/src/contracts/deposit.sol:442]: Note that the function is `public`, i.e. it's not only intended for internal use when we emit the events.

Comment (JamesHinshelwood) [reply zilliqa/src/contracts/deposit.sol:474]: Oh yes, I misunderstood the point of this function. Fair enough :)

**Review by JamesHinshelwood — APPROVED** [review]

**Pull request merged**